### DISCUSSION TITLE
How do I use an alternate ignore file with the new flat file config?

### DISCUSSION THREAD

**Discussion by askmrsinh** [eslint/eslint#18518] [Q&A] [answered]
I do not want to maintain two separate files `.gitignore` and `.eslintignore`. For most of my projects they are identical. [Earlier](https://eslint.org/docs/latest/use/configure/ignore-deprecated#using-an-alternate-file) I could just do 

```
npx eslint --ignore-path .gitignore .
```

This would ignore files that are ignore by Git (`.gitignore`). I am [unable to find a comparable config option in the CLI or Flat Config file](https://eslint.org/docs/latest/use/configure/ignore) to do this. Seem like a step backwards.

Am I missing something?

**Comment by askmrsinh** [comment]
The documentation should make it abundantly clear if there's no ready-made option to do this with the newer Flat File config. Otherwise people will just waste time trying to replicate the old `--ignore-path` switch.

Here's what I have settled on:

```js
import fs from 'node:fs/promises';
import globals from 'globals';
import pluginJs from '@eslint/js';
import tseslint from 'typescript-eslint';
import eslintPluginPrettierRecommended from 'eslint-plugin-prettier/recommended';

export default [
  {
    ignores: await fs
      .readFile('.gitignore', { encoding: 'utf8' })
      .then((r) => r.split(/[\r\n]+/).filter((r) => !r.trim().startsWith('#') && r.trim() !== ''))
  },
  { languageOptions: { globals: globals.browser } },
  pluginJs.configs.recommended,
  ...tseslint.configs.recommended,
  eslintPluginPrettierRecommended
];

```

**Reply by nzakas** [reply]
Agreed. Please feel free to open a PR on the documentation page you think this belongs on.

**Reply by askmrsinh** [reply]
Hey @nzakas, thank you for responding. I have identified the doc where such information would be useful and can make a PR. However, IMO this is a common use case; how do you feel about adding a new config option (or modifying how `ignore` works) to accommodate custom ignore filename/path? I can make a PR.

**Reply by nzakas** [reply]
We are pretty set on how we are doing ignores in the new config system, so I'm not on board with making any changes to that. If you'd like to solve the problem in code, I'd suggest that you create a utility than can read an ignore file and output `ignores` patterns, as you have in your code. Note that `.gitignore` patterns are different than Minimatch patterns, which is what ESLint uses, so you'll need to modify the patterns a bit to have them match appropriately.

**Reply by askmrsinh** [reply, marked as answer]
Glad to see a solution with the [@eslint/compat](https://www.npmjs.com/package/@eslint/compat) package: https://eslint.org/docs/latest/use/configure/ignore#including-gitignore-files. Making as answered.
